packages/glsp-sprotty/src/api-override.ts

```diff
@@ -198,3 +199,53 @@ export interface IActionDispatcher extends SIActionDispatcher {
 }
 
 export type IActionDispatcherProvider = () => Promise<IActionDispatcher>;
+
+/**
+ * Data structure that is passed to the `onCommandExecuted` event.
+ */
+export interface CommandExecutionData {
+    /** The command that has been executed successfully */
+    command: ICommand;
+    /** The new model root after the command execution */
+    newRoot: GModelRoot;
+}
+
+export interface ICommandStack extends SICommandStack {
+    /**
+     * Executes the given command on the current model and returns a
+     * Promise for the new result.
+     *
+     * Unless it is a special command, it is pushed to the undo stack
+     * such that it can be rolled back later and the redo stack is
+     * cleared.
+     */
+    execute(command: ICommand): Promise<GModelRoot>;
+
+    /**
+     * Executes all of the given commands. As opposed to calling
+     * execute() multiple times, the Viewer is only updated once after
+     * the last command has been executed.
+     */
+    executeAll(commands: ICommand[]): Promise<GModelRoot>;
+
+    /**
+     * Client-side undo/redo is not supported in GLSP. The server is responsible for handling undo/redo requests.
+     * This method is required to maintain compatibility with the sprotty API.
+     * Implementation should always be a no-op that returns the current model.
+     */
+    undo(): Promise<GModelRoot>;
+
+    /**
+     * Client-side undo/redo is not supported in GLSP. The server is responsible for handling undo/redo requests.
+     * This method is required to maintain compatibility with the sprotty API.
+     * Implementation should always be a no-op that returns the current model.
+     */
+    redo(): Promise<GModelRoot>;
+
+    /**
+     * Event that is fired after a command has been successfully executed on the stack. (i.e. after the model has been updated).
+     */
+    onCommandExecuted: Event<CommandExecutionData>;
+}
+
+export type CommandStackProvider = () => Promise<ICommandStack>;
```

Comment:
nit: Should we re-add the comments that is in the original code?

Actually, we do not use that interface in our code base at all and also do not offer any default binding for it [like Sprotty does](https://github.com/eclipse-sprotty/sprotty/blob/f9b2433481cc27a1ac0c92d525a92039ae7f6c76/packages/sprotty/src/base/di.config.ts#L81). Should we even export this, should we provide a default binding?